File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/event/transfer/MoneyTransferCompletedEvent.java
package org.jmolecules.example.axonframework.domain.event.transfer;

import org.axonframework.serialization.Revision;
import org.jmolecules.event.annotation.DomainEvent;

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "MoneyTransferCompletedEvent")
public record MoneyTransferCompletedEvent(
  String moneyTransferId,
  String sourceAccountId,

  int amount
) {

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/event/transfer/MoneyTransferCompletedEvent.kt
package org.jmolecules.example.axonframework.domain.event.transfer

import org.axonframework.serialization.Revision
import org.jmolecules.event.annotation.DomainEvent
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.Amount
import org.jmolecules.example.axonframework.domain.api.type.MoneyTransferId

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "MoneyTransferCompletedEvent")
data class MoneyTransferCompletedEvent(
    val moneyTransferId: MoneyTransferId,
    val sourceAccountId: AccountId,
    val amount: Amount
)
